### ISSUE TITLE
hashbang support!?

### ISSUE THREAD

**Issue by antoinegrant** [remix-run/react-router#601]
Did I miss something or there are no support for hashbang?

If it's not supported, is it planed to add the ability to chose to use a hashbang or a simple hash?

Thanks

**Comment by Feawel** [comment]
If definitely needed, you can use following stackoverflow answer http://stackoverflow.com/questions/25086832/how-to-stop-in-browser-with-react-router/32122395#32122395

@gpbl : one example where i needed it
  - front app with react on AWS s3 + cloudfront to allow max performance and no load problem
  - backend app only use as API endpoint
  - IE8 support :)

**Comment by kfuchs** [comment]
Agreed with @Feawel, totally needed.

**Comment by heybenji** [comment]
+1 for hashbang and thusly Google crawler support in #602. Also has added benefit of making URLs shareable on Facebook. If #602 had tests written, would the maintainers consider that pull request... Or are @aliceriot and I barking up the wrong tree

**Comment by taion** [comment]
If you want shareability and crawl-ability, you're going to need some sort of server-side support anyway, no? Facebook isn't going to request with the hash.

I'm not 100% opposed to adding this as a feature, maybe, but it seems like you're doing something that's quite contrary to modern best practices if you're trying to work that way.

**Comment by tvervest** [comment]
:+1: on hashbang support.

Like @Feawel, I simply need the feature for legacy support as I'm working on a new front-end on top of a legacy API and I'm bound to S3. I understand that the better solution is to use server-side rendering etc, but that simply isn't an option in this project. It's an unfortunate reality, but not all companies can be as future-ready as facebook ;)

**TL;DR**: Because of the lack of hashbang support I can't use react-router, please fix!

**Comment by taion** [comment]
If you want this feature, make an appropriate PR to history.

**Comment by tvervest** [comment]
PR #602 has already been rejected with this feature

**Comment by taion** [comment]
#602 had no tests and also isn't merge-able any more.

**Comment by tvervest** [comment]
Small update for those who run into this issue on Amazon S3; the problem can be circumvented by specifying the main `index.html` as the error page. It's not a perfect solution, but it's a good enough workaround ;)

More details are available in [this blog post](http://blog.boushley.net/2015/10/29/html5-deep-link-on-amazon-s3/).
